### DISCUSSION TITLE
Conditional routing?

### DISCUSSION THREAD

**Discussion by laneme** [vercel/vercel#4650] [Help] [answered]
I was wondering if i can rewrite the routes depending on certain conditions. Like if a header is set to this return that, otherwise the default one. Or any other sort of url parameters can used as conditions.
Is this currently possible? fastly (a edge cdn) offers similar conditional routing, for example.

**Comment by styfle** [comment, marked as answer]
Yes, you can read headers in your Serverless Function and respond differently based on the value and even cache the response if you wish.

For example:

```js
module.exports = (req, res) => {
  if (req.headers.origin = 'https://example.com') {
    res.json({ message: 'Hello, you are visiting from example.com' });
  } else {
    res.json({ message: 'Hello World' });
  }
}
```

- https://vercel.com/docs/v2/serverless-functions/introduction
- https://vercel.com/docs/v2/edge-network/caching
- https://vercel.com/edge-network
